libflux/core/src/parser/tests.rs

```diff
@@ -3888,6 +3888,203 @@ fn declare_variable_as_an_empty_array() {
     )
 }
 
+#[test]
+fn parse_empty_dict() {
+    let mut p = Parser::new(r#"[:]"#);
+    let parsed = p.parse_expression();
+    let loc = Locator::new(&p.source[..]);
+    assert_eq!(
+        parsed,
+        Expression::Dict(Box::new(DictExpr {
+            base: BaseNode {
+                location: loc.get(1, 1, 1, 4),
+                ..BaseNode::default()
+            },
+            elements: vec![],
+            lbrack: None,
+            rbrack: None,
+        }))
+    )
+}
+
+#[test]
+fn parse_single_element_dict() {
+    let mut p = Parser::new(r#"["a": 0]"#);
+    let parsed = p.parse_expression();
+    let loc = Locator::new(&p.source[..]);
+    assert_eq!(
+        parsed,
+        Expression::Dict(Box::new(DictExpr {
+            base: BaseNode {
+                location: loc.get(1, 1, 1, 9),
+                ..BaseNode::default()
+            },
+            elements: vec![DictItem {
+                key: Expression::StringLit(StringLit {
+                    base: BaseNode {
+                        location: loc.get(1, 2, 1, 5),
+                        ..BaseNode::default()
+                    },
+                    value: "a".to_string(),
+                }),
+                val: Expression::Integer(IntegerLit {
+                    base: BaseNode {
+                        location: loc.get(1, 7, 1, 8),
+                        ..BaseNode::default()
+                    },
+                    value: 0,
+                }),
+                comma: None,
+            }],
+            lbrack: None,
+            rbrack: None,
+        }))
+    )
+}
+
+#[test]
+fn parse_multi_element_dict() {
```

Comment:
Could you add a test were we parse the key as an expression: Something like `[1+1: 2]` or `[a:2]` etc